pkg/storage/etcd/etcd_watcher.go

```diff
@@ -398,10 +404,14 @@ func (w *etcdWatcher) sendModify(res *etcd.Response) {
 	var oldObj runtime.Object
 	if res.PrevNode != nil && res.PrevNode.Value != "" {
 		// Ignore problems reading the old object.
-		if oldObj, err = w.decodeObject(res.PrevNode); err == nil {
+		if oldObj, err = w.uncachedDecodeObject(res.PrevNode); err == nil {
 			oldObjPasses = w.filter(oldObj)
 		}
+		if err := w.versioner.UpdateObject(oldObj, res.Node.ModifiedIndex); err != nil {
```

Comment:
Yeah, that process logic seems a tad questionable. Let's see what @wojtek-t
thinks.

On Wed, May 11, 2016 at 8:08 PM, Hongchao Deng <notifications@github.com>
wrote:

> In pkg/storage/etcd/etcd_watcher.go
> <https://github.com/kubernetes/kubernetes/pull/25415#discussion_r62959506>
> :
>
> >  			oldObjPasses = w.filter(oldObj)
> >  		}
> > +		if err := w.versioner.UpdateObject(oldObj, res.Node.ModifiedIndex); err != nil {
>
> it should send the RV corresponding with the object state
>
> I am supportive of this argument.
>
> There is one problem if we change sendDelete to set the corresponding
> (old) RV: in cacheWatcher.process()
> <https://github.com/kubernetes/kubernetes/blob/995f022808d1407897a5b23373484ac683534390/pkg/storage/cacher.go#L579-L581>,
> it only sends larger resource version. Maybe we should fix that and all
> assumptions made upon the event from sendDelete(). Right?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25415/files/4000ce3894c2f9e548093c223c5890aa26a96aa4#r62959506>
>